src/crypt/decrypter.ts

```diff
@@ -163,15 +163,23 @@ export default class Decrypter {
         return crypto.decrypt(data.buffer, aesKey);
       })
       .catch((err) => {
-        return this.onWebCryptoError(err, data, key, iv) as ArrayBuffer;
+        logger.warn(
+          '[decrypter.ts]: WebCrypto Error, disable WebCrypto API:',
+          err
+        );
+        return this.onWebCryptoError(data, key, iv);
       });
   }
 
-  private onWebCryptoError(err, data, key, iv): ArrayBuffer | null {
-    logger.warn('[decrypter.ts]: WebCrypto Error, disable WebCrypto API:', err);
+  private onWebCryptoError(data, key, iv): ArrayBuffer {
     this.config.enableSoftwareAES = true;
     this.logEnabled = true;
-    return this.softwareDecrypt(data, key, iv);
+    const result = this.softwareDecrypt(data, key, iv);
+    if (result === null) {
+      this.reset();
+      throw new Error(`softwareDecrypt: result is 'null'`);
+    }
+    return result;
```

Comment:
For `softwareDecrypt` to work we actually have to call flush (see line 91 in `decrypt` for reference):

```suggestion
    this.softwareDecrypt(data, key, iv);
    const decryptResult = this.flush();
    if (decryptResult) {
      return decryptResult.buffer;
    }
    this.reset();
    throw new Error(`softwareDecrypt: result is 'null'`);
```

Since we're throwing here, we should make the return type `ArrayBuffer | never`.

Reply:
Here I agree 100%, I found a comment in another file. 
``` 
// Software decryption is progressive. Progressive decryption may not return a result on each call. Any cached
// data is handled in the flush() call
```

https://github.com/video-dev/hls.js/blob/8f55f8507a0aff02e0251218b46765ee1a468955/src/crypt/decrypter.ts#L174